spec/index.md

```diff
@@ -3745,6 +3804,8 @@ Depending on whether this is a call message and a response messages, we have eit
 
 The cycle consumption of executing this message is modeled via the unspecified `cycles_used` variable; the variable takes some value between 0 and `MAX_CYCLES_PER_MESSAGE`/`MAX_CYCLES_PER_RESPONSE` (for call execution and response execution, respectively).
 
+The logs produced by the canister during message execution are modeled via the unspecified `canister_logs` variable; the variable stores a list of logs (each of type `CanisterLog`) with consecutive sequence numbers, timestamps equal to `S.time[M.receiver]`, and contents produced by the canister calling `ic0.debug_print`, `ic0.trap`, or produced by the WebAssembly runtime when the canister WebAssembly module traps.
```

Comment:
Does `ic0.trap` correspond to `HypervisorError::CalledTrap(text)` and "produced by the WebAssembly runtime when the canister WebAssembly module traps" correspond to `HypervisorError::Trapped(trap_code)` in the replica code? What does a panic in unwrapping in Rust canister code correspond to then?

Reply:
in my early testing I had different prefixes for `CalledTrap` and `Trapped`, it was `Canister explicitly called trap:` and `Canister trapped:` correspondingly. So panic was `CalledTrap`, see https://github.com/maksymar/canister-logs:

```shell
... Canister explicitly called trap: Panicked at 'aaa!', demo_canister/src/lib.rs:52:5
```

Reply:
> So panic was CalledTrap

That's interesting - a panic in Rust code results in `ic0.trap`? How does the Rust compiler know about that?

Reply:
Thank you for the explanation!